gtdynamics/universal_robot/Joint.cpp

```diff
@@ -357,6 +314,60 @@ gtsam::Vector6_ Joint::twistConstraint(uint64_t t) const {
   return twist_c_hat - twist_c;
 }
 
+/* ************************************************************************* */
+gtsam::Vector6_ Joint::twistAccelConstraint(uint64_t t) const {
+  gtsam::Vector6_ twist_c(internal::TwistKey(child()->id(), t));
+  gtsam::Vector6_ twistAccel_p(internal::TwistAccelKey(parent()->id(), t));
+  gtsam::Vector6_ twistAccel_c(internal::TwistAccelKey(child()->id(), t));
+  gtsam::Double_ q(internal::JointAngleKey(id(), t));
+  gtsam::Double_ qVel(internal::JointVelKey(id(), t));
+  gtsam::Double_ qAccel(internal::JointAccelKey(id(), t));
+
+  /// The following 2 lambda functions computes the expected twist acceleration
+  /// of the child link. (Note: we split it into 2 functions because the
+  /// expression constructor currently only support up to tenary expression.)
+  auto transformTwistAccelTo1 =
+      [this](double q, const Vector6 &other_twist_accel,
+             gtsam::OptionalJacobian<6, 1> H_q,
+             gtsam::OptionalJacobian<6, 6> H_other_twist_accel) {
+        Pose3 jTi = relativePoseOf(parent(), q);
+        Vector6 this_twist_accel = jTi.AdjointMap() * other_twist_accel;
+
+        if (H_other_twist_accel) {
+          *H_other_twist_accel = jTi.AdjointMap();
+        }
+        if (H_q) {
+          // TODO(frank): really, zero below? Check derivatives. Also,
```

Comment:
Create issue to check this (or solve it by checking derivatives)

Reply:
The derivatives are checked in testTwistAccelerationFactor. So it should be fine.